### PR TITLE
Rename DTK Search to ArborX

### PR DESCRIPTION
**Pull request by aprokop**

### CODE REVIEW

**Comment by aprokop** [comment]
Do we simply change the license too to "ArborX authors". Is everyone OK with it? @sslattery?

**Comment by sslattery** [comment]
I'm fine with that

File: packages/Discretization/src/DTK_Interpolation_decl.hpp [hunk 1]
```diff
@@ -13,12 +13,12 @@
 #define DTK_INTERPOLATION_DECL_HPP
 
 #include "DTK_ConfigDefs.hpp"
+#include <ArborX.hpp>
```
Comment (dalg24) [inline packages/Discretization/src/DTK_Interpolation_decl.hpp]: Please include this header after the DTK block.

Comment (dalg24) [reply packages/Discretization/src/DTK_Interpolation_decl.hpp]: Same comment everywhere below.

File: packages/Search/test/tstDetailsTreeTraversal.cpp [hunk 2]
```diff
@@ -21,19 +21,21 @@
...
 
-#include "DTK_EnableViewComparison.hpp"
+#include "ArborX_EnableViewComparison.hpp"
 
 #include <boost/test/unit_test.hpp>
 
-namespace dtk = DataTransferKit::Details;
+#define BOOST_TEST_MODULE DetailsTreeTraversal
+
+namespace details = ArborX::Details;
```
Comment (dalg24) [inline packages/Search/test/tstDetailsTreeTraversal.cpp]: How about a using-directive?

File: packages/Discretization/src/DTK_DiscretizationHelpers.hpp [hunk 3]
```diff
@@ -120,7 +120,7 @@ void computeOffset( Kokkos::View<T1 *, DeviceType> predicate, T2 value,
                           } );
```
[the remark this answers is not in the document]
Comment (aprokop) [reply packages/Discretization/src/DTK_DiscretizationHelpers.hpp]: Won't fix right now. It compiles, and that is the only criteria.

File: packages/Discretization/src/DTK_Interpolation_decl.hpp (hunk 1 above)
Comment (aprokop) [reply packages/Discretization/src/DTK_Interpolation_decl.hpp]: See above.

File: packages/Search/test/tstDetailsTreeTraversal.cpp (hunk 2 above)
Comment (aprokop) [reply packages/Search/test/tstDetailsTreeTraversal.cpp]: Out of scope of this PR, can be addressed later.

**Pull request merged**